### PR TITLE
drivers: serial: Update drivers to use devicetree Kconfig symbol

### PR DESCRIPTION
**Pull request by galak**
Update serial drivers to use DT_HAS_<compat>_ENABLED Kconfig symbol
to expose the driver and enable it by default based on devicetree.

We remove 'depend on' Kconfig for symbols that would be implied by
the devicetree node existing.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

File: drivers/serial/Kconfig.gecko [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 config UART_GECKO
 	bool "Gecko UART/USART driver"
-	depends on HAS_SILABS_GECKO
-	depends on GPIO_GECKO
+	default y
+	depends on DT_HAS_SILABS_GECKO_USART_ENABLED
```
Comment (str4t0m) [inline drivers/serial/Kconfig.gecko:9]: I believe this should be enabled for DT_HAS_SILABS_GECKO_USART_ENABLED as well as DT_HAS_SILABS_GECKO_UART_ENABLED

File: drivers/serial/Kconfig.pl011 [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 menuconfig UART_PL011
 	bool "ARM PL011 UART Driver"
+	default y
```
Comment (str4t0m) [inline drivers/serial/Kconfig.pl011:6]: In this file UART_PL011_SBSA likely should be enabled in case DT_HAS_ARM_SBSA_UART_ENABLED

**Review by str4t0m — CHANGES_REQUESTED** [review]

File: drivers/serial/Kconfig.test [hunk 3]
```diff
@@ -6,6 +6,8 @@ DT_COMPAT_VND_SERIAL := vnd,serial
 # Hidden option to enable the vnd,serial dummy driver used in testing.
 config SERIAL_TEST
 	def_bool $(dt_compat_enabled,$(DT_COMPAT_VND_SERIAL))
```
[the remark this answers is not in the document]
Comment (galak) [reply drivers/serial/Kconfig.test:8]: it is.  Will fix.

**Comment by galak** [comment]
@dcpleung @str4t0m take another look.

**Review by dcpleung — DISMISSED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]
LGTM

**Pull request merged**